examples/tcp_echo_multi_server.rs

```diff
@@ -0,0 +1,73 @@
+//! This is a simple TcpServer example using spawn_link! macro
+//!
+//! 1) Without Supervisor or AbstractProcess
+//! 2) Using Serde passed structs
+//! 3) To implement simple linebuffer echo back
+//! 4) With ability to listen on multiple local addresses
+//!
+use lunatic::{net, sleep, spawn_link, Mailbox};
+use std::io::{BufRead, BufReader, Write};
+use std::time::Duration;
+
+#[derive(serde::Serialize, serde::Deserialize)]
+struct TcpServer {
+    local_address: String,
+}
+
+impl TcpServer {
+    fn new(local_address: &str) -> Self {
+        TcpServer {
+            local_address: local_address.to_owned(),
+        }
+    }
+}
+
+#[derive(serde::Serialize, serde::Deserialize)]
+struct TcpPeer {
+    tcp_stream: lunatic::net::TcpStream,
+    peer: std::net::SocketAddr,
+}
+
+#[lunatic::main]
+fn main(_: Mailbox<()>) {
+    let local_addrs = vec![
+        TcpServer::new("127.0.0.1:6666"),
+        TcpServer::new("127.0.0.1:6667"),
+    ];
+
+    for l in local_addrs {
+        spawn_link!(|input = l| listen(input));
+    }
+
+    loop {
+        // Mainloop every 10s
+        sleep(Duration::from_millis(10000));
+    }
```

Comment:
A common pattern in lunatic is just to define a mailbox in the main process and block forever by waiting on a message that will not arrive. Something like:

```rust
fn main(mailbox: Mailbox<()>) {
    // ...

    // This will block forever
    let _ = mailbox.receiver();
}
```

Reply:
Makes sense. I'm too used to being :sleeping:  Thanks!